### PR TITLE
Refactor the team collection, post.authors, and related implementations

### LINKED ISSUE #242: Refactor post.authors and related implementations
Right now, `post.authors` is a list of objects with two attributes: `fname` and `lname`. This nested structure makes it troublesome to access the list of authors' names of every post in an elegant way. Specifically, it caused some tricky problems in issues like #213 and #163.

A possible solution is to refactor post.authors into a list of `fname` + `lname` strings, instead of the current nested structure. This might also help with future implementations that involve accessing authors of posts, like #245  :)

### PR DESCRIPTION
**Pull request by MDZHX**
### Closes #242
- Combined `firstname` and `lastname` fields of the `team` collection, preserved the original style of quotation marks of each team member's `.md` file.
- Combined `fname` and `lname` in `post.authors`.
- Updated related files to be compatible with changes: `team.html`, `team-member.html`, `post.html`.
- Updated `README.md` for the `team` collection

### Changes unrelated to #242
- Thanks to the refactor, `team-member.html` now uses a cleaner way to create an iterable of all blog posts of a member.
- Team member pages now displays "Blog posts that MEMBER have/has written: " depending on the pronouns.
- Change the div structure in `team-member.html` so that blog posts become a separate <div> `team-posts`, instead of being inside of the <div> `team-profile`

### Known Problem

- [x]  `team.html` used to sort members using the `lastname` field when creating iterables. Now without the `lastname` field, members aren't properly sorted.

### Potential Problems

- [x] The statements checking for pronouns could be flawed / not clean enough.
- [x] Although I checked through files to find those that might be impacted by this refactoring, there might be files that I missed. **e.g.** I updated the README for `team`, but I didn't find a README for `post`, which has also been refactored.
- [x] In team member pages, the whitespace above and below the line "Blog posts that MEMBER have/has written:" seems a bit unbalanced.

### CODE REVIEW

File: _layouts/team-member.html [hunk 1]
```diff
@@ -48,20 +48,20 @@ <h1 class="title">{{ page.firstname }} {{ page.lastname }}
...
+<div class="team-posts">
+    {% assign authors_posts = site.posts | where_exp:"post", "post.authors contains page.name" | sort_by:"date" %}
+    {% if authors_posts.size > 0 %}
+        {% assign firstname = page.name | split: " " | first %}
+        {% if page.pronouns and page.pronouns contains "they" %}
+            {% assign have_has = "have" %}
+        {% else %}
+            {% assign have_has = "has" %}
+        {% endif %}
+        <h2 class="title"> Blog posts that {{ firstname }} {{ have_has }} written: </h2>
```
Comment (malsf21) [inline _layouts/team-member.html:63]: I think this is my only nitpick; regardless of pronouns (e.g. they/them), I think we'd still use has.

For example:
* Blog posts that Matt has written (he/him)
* Blog posts that Sophie has written (she/her)
* Blog posts that Arjun has written (they/them)

So, no need for the `have_has` variable, though I appreciate you thinking about this!

**Review by malsf21 — CHANGES_REQUESTED** [review]
Wow Zihan, this is looking great! Overall, this is a great (and much-needed refactor), and it'll make our lives significantly easier when we implement new features 😊 

Only two small notes:

* I appreciate you considering they/them pronouns in rendering! In this case, it's still they/them singular, so we'd use "has" no matter what; I do like your solution there though!
* thanks for updating the documentation in `_team/README.md`; you'll also need to update the accompanying docs in the (poorly named) `_docs/static-setup.md`

Other than that, I think this is good to merge! This is really wonderful work, I think after you wrap up everything related to this (e.g. the More Info section), we'll showcase your work on our blog!

**Comment by malsf21** [comment]
Oops, so sorry, forgot to address your questions. I love that you outlined potential and known problems, it makes my life as a code reviewer so much easier! To just address each of the points:

> - [ ]  `team.html` used to sort members using the `lastname` field when creating iterables. Now without the `lastname` field, members aren't properly sorted.

Great point! I think it's fine to sort by the overall name; this functionally sorts by first name, which if anything might be easier for kids to find.

> - [ ] The statements checking for pronouns could be flawed / not clean enough.

Addressed in above comment; we don't need to be checking for pronouns.

> - [ ] Although I checked through files to find those that might be impacted by this refactoring, there might be files that I missed. **e.g.** I updated the README for `team`, but I didn't find a README for `post`, which has also been refactored.

Great point! I pointed out an area where you may have missed some documentation, but passing the compilation means that we've done all of our field lookups properly. Ctrl+F on an entire project here is your friend!

> - [ ] In team member pages, the whitespace above and below the line "Blog posts that MEMBER have/has written:" seems a bit unbalanced.

Yep, you can either choose to resolve that in this PR, or in a follow-up. Don't overstress about this one, it's a small fix!

**Comment by MDZHX** [comment]
Thanks for the careful review! I'll soon address the pronoun/doc/sorting issues :)

Also just to confirm, for the whitespace problem, am I supposed to adjust some of the `.scss` files? (I'm not quite familiar with this since I haven't done much styling-related things before.)

**Comment by malsf21** [comment]
> Also just to confirm, for the whitespace problem, am I supposed to adjust some of the `.scss` files? (I'm not quite familiar with this since I haven't done much styling-related things before.)

Yup, exactly! You'll likely find most of the relevant classes in `_sass/_basic.scss`.

**Review by malsf21 — APPROVED** [review]
Zihan,

I've got to say - this stuff looks absolutely amazing. I really appreciate how in-depth your pull request review has been - it makes code review for me very easy.

Things look good to merge on my end. Please Squash & Merge this PR too!

**Pull request merged**